src/js/view/components/timeslider.js

```diff
@@ -157,11 +157,22 @@ define([
             var pct = position / _railBounds.width;
             var time = duration * pct;
 
+            // For DVR we need to swap it around
+            if (duration < 0) {
+                time = duration - time;
+            }
+
             var timetipText;
             if (this.activeCue) {
                 timetipText = this.activeCue.text;
             } else {
-                timetipText = utils.timeFormat(time);
+                var allowNegativeTime = true;
+                timetipText = utils.timeFormat(time, allowNegativeTime);
+
+                // If DVR and within one second of live
+                if (duration < 0 && (Math.abs(time) < 1) ) {
```

Comment:
Does this work with DVR stream time events?

"within one second of live" `if (duration < 0 && (Math.abs(time) < 1)` 

I would expect it to be more like within 25-30 seconds is "Live" and 1 second of latency is very rare or unlikely.